CMakeLists.txt

```diff
@@ -148,14 +149,33 @@ if (NOT "$ENV{HOROVOD_WITHOUT_MPI}" STREQUAL "1")
 endif()
 
 # CUDA and ROCM
+set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
+if(NOT DEFINED CMAKE_CUDA_RUNTIME_LIBRARY)
+    set(CMAKE_CUDA_RUNTIME_LIBRARY "Shared")  # Set to "Static" or "Shared"
+endif()
+if (DEFINED ENV{HOROVOD_CUDA_HOME})
+    set(CMAKE_CUDA_COMPILER $ENV{HOROVOD_CUDA_HOME}/bin/nvcc)
+endif()
+include(CheckLanguage)
+check_language(CUDA)
+if (CMAKE_CUDA_COMPILER)
+    if ((CMAKE_CXX_COMPILER_ID MATCHES GNU) AND (CMAKE_SYSTEM_PROCESSOR MATCHES ppc64le))
+        if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8.0)
+            set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -std=c++11")
```

Comment:
Why is that needed if in `horovod/common/ops/cuda/CMakeLists.txt`, you already set:
```
set(CMAKE_CUDA_STANDARD 11)
```
?

Reply:
Hi @nvcastet, thanks for the review and all the advice earlier!

The `-std=c++11` flag here is for `enable_language(CUDA)` in this top-level `CMakeLists.txt`. CMake will apparently compile a test program at that point to gauge whether the compiler is really set up correctly etc. That fails on ppc64le with our versions of gcc and CUDA, however, because of a float128-related bug and one way to circumvent that is to disable C++14 support. I tried `set(CMAKE_CUDA_STANDARD 11)` first to achieve that, but that setting is apparently ignored at this stage, so we got the same error: [Jenkins log](https://powerci.osuosl.org/job/Horovod_PPC64LE_GPU_PIPELINE/view/change-requests/job/PR-3261/14/console), [intermediate commit](https://github.com/maxhgerlach/horovod/commit/5d758a5e02768435ce63db2afeacf6a8188fa861). In contrast, `CMAKE_CUDA_FLAGS` is not ignored there. I don't know why.